addons/mail/models/mail_thread.py

```diff
@@ -2274,6 +2274,28 @@ def _message_compute_parent_id(self, parent_id):
                 parent_id = current_ancestor.parent_id.id if current_ancestor.parent_id else parent_id
         return parent_id
 
+    def _message_compute_subject(self, record_names=None, model_description=None):
+        """ Helper to get the default subject when creating a message on this thread
+            For most records it should just be the name of the record,
+            override in child model when this does not make sense
+            :param record_name: {thread_id -> record_name} for records which were already computed
+            :return: {thread_id -> record_name} for every records
+        """
+        record_names = dict(record_names) if record_names else dict()
+
+        missing_names = self.filtered(lambda rec: not record_names.get(rec.id)).name_get()
+        record_name_gets = {key: value for key, value in missing_names if value}
```

Comment:
You could actually call dict() directly on messing_names, as it is a list of tuples, it should work automatically :)